toast-tk-web-agent/src/main/java/io/toast/tk/agent/run/TestRunner.java

```diff
@@ -0,0 +1,104 @@
+package io.toast.tk.agent.run;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.util.Strings;
+
+import com.google.inject.Module;
+
+import io.toast.tk.agent.config.AgentConfigProvider;
+import io.toast.tk.agent.ui.NotificationManager;
+import io.toast.tk.dao.domain.impl.test.block.ITestPage;
+//import io.toast.tk.dao.domain.impl.test.block.ITestPlan;
+import io.toast.tk.plugin.IAgentPlugin;
+import io.toast.tk.plugin.PluginLoader;
+import io.toast.tk.runtime.parse.FileHelper;
+import io.toast.tk.runtime.parse.TestParser;
+
+public class TestRunner {
+	private static final Logger LOG = LogManager.getLogger(TestRunner.class);
+	private TestPageRunner testPageRunner;
+	private AgentConfigProvider provider;
+	public String fileName;
+	
+	private boolean interupted = false;
+	
+	public TestRunner(AgentConfigProvider provider){
+		this.provider = provider;
+	}
+	
+	public void execute(){
+		String scriptsPath = this.provider.get().getScriptsDir();
+		Path path = Paths.get(scriptsPath);
+		if(!Strings.isEmpty(scriptsPath) || !Files.isDirectory(path)){
+			NotificationManager.showMessage("No script directory defined, check agent settings !");
+		}
+		
+		final List<ITestPage> testScripts = getScripts(path);
+		
+		executeScripts(testScripts);
+	}
+	
+	private List<ITestPage> getScripts(Path path) {
+		List<ITestPage> testScripts = new ArrayList<>();
+		TestParser parser = new TestParser();
+		try {
+			Files.list(path).forEach(p -> {
+				try{
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1141)